### ISSUE TITLE
purchaseUpdateListener  for iOS subscription is always called twice

### ISSUE THREAD

**Issue by Saikedo** [dooboolab-community/react-native-iap#2713]
**Description**

When making purchases or receiving renewal updates through purchaseUpdateListener on iOS, the notification always arrives twice for some reason. They always arrive somewhere around 2.5 seconds after each other. I tried calling finishTransaction right after I receive the purchaseUpdateListener but still nothing, it works the same way.

Any idea what might be the reason for this double notification for the same purchase?

I am using the Storekit 2 in case if that is relevant.

**Expected Behavior**

purchaseUpdateListener  should have fired only once for the same purcahse

**Environment:**

- react-native-iap: 12.13.0
- Platforms: iOS

**Comment by harrymash2006** [comment]
I am also facing the same issue. Did you get any solution?

**Comment by Saikedo** [comment]
No, I also tried with a minimal example and it is still happening. For now I just made sure that the extra call is not causing any issues on frontend or backend at least so everything is handled correctly even if it gets called multiple times but the double call was still bothering me a bit.

**Comment by PatriciaSauer** [comment]
How do you make sure that the extra call is not causing any issues? I am experiencing this on Android

**Comment by Karthik-V26** [comment]
Hi @Saikedo , @harrymash2006 ,
Any update on this?
how you handled this issue?

**Comment by RomanKurilovich** [comment]
Hi @Karthik-V26 , @FXStreetDeveloper .
Any update on this?
how you handled this issue?

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
